pkg/client/cache/reflector.go

```diff
@@ -277,6 +278,7 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			if urlError, ok := err.(*url.Error); ok {
 				if opError, ok := urlError.Err.(*net.OpError); ok {
 					if errno, ok := opError.Err.(syscall.Errno); ok && errno == syscall.ECONNREFUSED {
```

Comment:
We really want this logic to be abstracted - there are other errors that are equivalent.  Please move this into a public IsConnectionRefused in `pkg/client/unversioned`